dvc/remote/gs.py

```diff
@@ -136,16 +136,33 @@ def remove(self, path_info):
 
         blob.delete()
 
-    def _list_paths(self, bucket, prefix):
-        for blob in self.gs.bucket(bucket).list_blobs(prefix=prefix):
+    def _list_paths(self, bucket, prefix, max_items=None):
+        for blob in self.gs.bucket(bucket).list_blobs(
+            prefix=prefix, max_results=max_items
+        ):
             yield blob.name
 
     def list_cache_paths(self):
-        return self._list_paths(self.path_info.bucket, self.path_info.path)
+        return self.walk_files(self.path_info)
```

Comment:
Ooops, tests are failing due to this for `gs`. `walk_files` provides absolute urlpath, whereas `_list_paths` provided relative path.

Reply:
Fixed by 54dd9890dc83b

Reply:
It's failing again, needs more investigation.

Reply:
`tests/unit/test_ignore.py` seems to be always failing after running tests once. I have run all the tests 10 times, and it's always the 6 scenarios in that file that are failing. Other tests (with `DVC_TEST_GCP` are passing).

Reply:
@skshetry I'm not able to reproduce `test_ignore.py` failing on master or on your branch 🤔